langstream-runtime/langstream-runtime-impl/src/main/python/langstream/api.py

```diff
@@ -60,11 +61,25 @@ def headers(self) -> List[Tuple[str, Any]]:
 
 RecordType = Union[Record, dict, list, tuple]
 
+class AgentContext(ABC):
+    """The Agent context interface"""
+
+    def __init__(self):
+        """Initialize the agent context."""
+        pass
+
+
+    @abstractmethod
+    def get_persistent_state_directory(self):
+        """Return a path pointing to the stateful agent directory. Return None if not configured in the agent."""
+        pass
+
+
 
 class Agent(ABC):
     """The Agent interface"""
 
-    def init(self, config: Dict[str, Any]):
+    def init(self, config: Dict[str, Any], context: AgentContext):
```

Comment:
is this a breaking change ?

maybe we should have same approach as in Java and have a "setAgentContext" function

Reply:
no it is not, I added a test for the old syntax